services/api/service.go

```diff
@@ -635,6 +751,37 @@ func (api *RelayAPI) updateProposerDuties(headSlot uint64) {
 	api.log.Infof("proposer duties updated: %s", strings.Join(_duties, ", "))
 }
 
+func (api *RelayAPI) prepareOptimisticBuildersForSlot(headSlot uint64) {
```

Comment:
```suggestion
func (api *RelayAPI) prepareBuildersForSlot(headSlot uint64) {
```

this function is critical to prepare builder info even for non-optimistic flow. I argue it's a core function for any mode of relay operation, and we should remove optimistic from the name

Reply:
done!